src/Nethermind/Nethermind.Network.Discovery/Discv5/NethermindLoggerFactory.cs

```diff
@@ -0,0 +1,55 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Microsoft.Extensions.Logging;
+using Nethermind.Logging;
+
+namespace Nethermind.Network.Discovery;
+
+internal class NethermindLoggerFactory(ILogManager logManager, bool lowerLogLevel = false) : ILoggerFactory
+{
+    public Microsoft.Extensions.Logging.ILogger CreateLogger(string categoryName)
+    {
+        return new NethermindLogger(logManager.GetLogger(categoryName), lowerLogLevel);
+    }
+
+    public void Dispose() { }
+
+    public void AddProvider(ILoggerProvider provider) { }
+
+    class NethermindLogger(Logging.ILogger logger, bool lowerLogLevel = false) : Microsoft.Extensions.Logging.ILogger
+    {
+        public IDisposable? BeginScope<TState>(TState state) where TState : notnull => null;
+
+        public bool IsEnabled(Microsoft.Extensions.Logging.LogLevel logLevel) => true;
```

Comment:
This should correctly check if log level is enabled.

Reply:
Indeed